java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1928,11 +1928,37 @@ private static native long clamper(long nativeView, long loScalarHandle, long lo
   public ColumnVector matchesRe(String pattern) {
     assert type == DType.STRING : "column type must be a String";
     assert pattern != null : "pattern may not be null";
-    assert pattern.isEmpty() == false : "pattern string may not be empty";
+    assert !pattern.isEmpty() : "pattern string may not be empty";
     try (DevicePrediction prediction = new DevicePrediction(predictSizeFor(DType.BOOL8), "matchesRe")) {
       return new ColumnVector(matchesRe(getNativeView(), pattern));
     }
   }
+
+  /**
+   * Returns a boolean ColumnVector identifying rows which
+   * match the given regex pattern.
```

Comment:
I think your javadocs need to be updated for containsRe not matchesRe

Reply:
The difference here is : `but only at the beginning of the string.` Does that seem ok to you?